src/main/java/spireMapOverhaul/zones/brokenSpace/BrokenSpaceZone.java

```diff
@@ -0,0 +1,254 @@
+
+        // pick RANDOM rarity
+        int roll = rng.random(7);
+        switch (roll) {
+            case 0:
+            case 1:
+                rarity = AbstractCard.CardRarity.COMMON;
+                break;
+            case 2:
+            case 3:
+                rarity = AbstractCard.CardRarity.UNCOMMON;
+                break;
+            case 4:
+            case 5:
+                rarity = AbstractCard.CardRarity.RARE;
+                break;
+            case 6:
+                return AbstractDungeon.returnRandomCurse().makeCopy();
+
+        }
+
+
+        AbstractCard c = CardLibrary.getAnyColorCard(rarity).makeCopy();
+
+        return c;
+
+
+    }
+
+    @Override
+    public ArrayList<AbstractCard> getAdditionalCardReward() {
+        ArrayList<AbstractCard> cards = new ArrayList<>();
+        int amount = 3;
+
+        for (AbstractRelic r : AbstractDungeon.player.relics) {
+            amount = r.changeNumberOfCardsInReward(amount);
+        }
+
+        for (int i = 0; i < amount; i++) {
+            AbstractCard c = getTrulyRandomCard(AbstractDungeon.cardRandomRng);
+            cards.add(c);
+            UnnaturalCardField.unnatural.set(c, true);
+
+        }
+        return cards;
+    }
+
+    @Override
+    public void modifyReward(RewardItem rewardItem) {
+        if (rewardItem.type == RewardItem.RewardType.RELIC) {
+            AbstractRelic origRelic = rewardItem.relic;
+            AbstractRelic newRelic = getValidBrokenRelic();
+
+            if (newRelic != null) {
+                rewardItem.relic = newRelic;
+                rewardItem.text = newRelic.name;
+
+                switch (origRelic.tier) {
+                    case COMMON:
+                        AbstractDungeon.commonRelicPool.add(origRelic.relicId);
+                        break;
+                    case UNCOMMON:
+                        AbstractDungeon.uncommonRelicPool.add(origRelic.relicId);
+                        break;
+                    case RARE:
+                        AbstractDungeon.rareRelicPool.add(origRelic.relicId);
+                        break;
+                    case SHOP:
+                        AbstractDungeon.shopRelicPool.add(origRelic.relicId);
+                        break;
+                    case BOSS:
+                        AbstractDungeon.bossRelicPool.add(origRelic.relicId);
+                        break;
+                    default:
+                        SpireAnniversary6Mod.logger.info("what.");
+                        break;
+
+                }
+            }
+
+        }
+    }
+
+    public AbstractRelic getValidBrokenRelic() {
+        ArrayList<String> validRelics = new ArrayList<>();
+        for (String relicID : BrokenRelics) {
+            if (!adp().hasRelic(relicID)) {
+                validRelics.add(relicID);
+            }
+        }
+        if (validRelics.isEmpty()) {
+            return new Circlet();
+        }
+        AbstractRelic r;
```

Comment:
Why not just check if the relic can spawn while checking if the player has the relic?